fendermint/vm/message/src/signed.rs

```diff
@@ -33,6 +34,16 @@ pub enum SignedMessageError {
     Ethereum(#[from] anyhow::Error),
 }
 
+/// Domain specific transaction hash.
+///
+/// Some tools like ethers.js refuse to accept Tendermint hashes,
+/// which use a different algorithm than Ethereum.
+///
+/// We can potentially extend this list to include CID based indexing.
+pub enum DomainHash {
```

Comment:
I really like this abstraction, we may not use it ever, but it may also safe some headaches in the future

Reply:
Yeah, I think even a list might be possible. Like a commented I saw in Lotus it's possible to retrieve a message by CID, so who knows, the need might come.